lib/shelley/src/Cardano/Wallet/Shelley/Network/Node.hs

```diff
@@ -714,9 +716,9 @@ mkTipSyncClient tr np onPParamsUpdate onInterpreterUpdate onEraUpdate = do
                     <$> LSQry Byron.GetUpdateInterfaceState)
                 (fromShelleyPParams eraBounds ppNode
                     <$> LSQry Shelley.GetCurrentPParams)
-                (fromShelleyPParams eraBounds ppNode
+                (fromAllegraPParams eraBounds ppNode
                     <$> LSQry Shelley.GetCurrentPParams)
-                (fromShelleyPParams eraBounds ppNode
```

Comment:
Is there no simple way to keep using `fromShelleyPPararams` for Allegra and Mary rather than duplicating it?

I believe large amounts of duplication in "compatibility layers" has been a cause of bugs in the past. Though if we are moving to embrace cardano-api types soon anyway, it might not matter too much...

But shouldn't adding some `IsShelleyBasedEra era` constraint to `fromShelleyPParams` and then constructing the minimumUTxOValue there work?

Reply:
> Is there no simple way to keep using fromShelleyPPararams for Allegra and Mary rather than duplicating it?

This would also be my preference!

> ...But shouldn't adding some `IsShelleyBasedEra era` constraint to `fromShelleyPParams` and then constructing the minimumUTxOValue there work?

I did try this, but was unable to get the types to line up (though I can't remember the precise error). I'll give it another try!

Reply:
Hi @Anviking 

Just a thought:

Perhaps it's clearer as it currently is, with one separate function for each era?

Given that these functions are always called with a particular era in hand, it's hard for people to use the wrong function for a given era, since every function requires an era-specific `PParams` value.

Of course, we could try to merge `from{Shelley,Allegra,Mary}PParams` into one function, but what would we call the function?

* We could call it `fromShelleyBasedEraPParams`, but this name implies that it ought to be usable for any Shelley-based era, including those after `Mary`, which would be a mistake.
* We could call it `fromShelleyAllegraMaryPParams`, and this name would _perhaps_ be more accurate, but it seems rather awkward.